setup.py

```diff
@@ -1,10 +1,29 @@
-import os
 from setuptools import setup
-from setuptools.config import read_configuration
+import ast
+try:
+    from pathlib import Path
+except:
+    from pathlib2 import Path
 
-this_dir = os.path.dirname(__file__)
-cfg = read_configuration(os.path.join(this_dir, 'setup.cfg'))
-cfg["options"].update(cfg["metadata"])
-cfg = cfg["options"]
 
-setup(**cfg)
+def extract_global_vars(file):
+    t = file.read_text(encoding="utf-8")
+    m = ast.parse(t)
```

Comment:
I would read the file in binary mode instead of hardcoding the encoding here. `ast.parse` accepts `bytes` input, in which case it lets the parser determine the encoding (this also respects `# coding: ...` comments and such).

In general I'm in favor of "UTF-8 everywhere", but in this case we specifically want to behave like Python's normal parser, so we shouldn't override its defaults.